File: SharedCode/src/androidMain/kotlin/rxProxy/factories/SchedulerModuleImpl.kt
package com.jetbrains.handson.mpp.mobile.rxProxy.factories

import com.jetbrains.handson.mpp.mobile.rxProxy.Scheduler

class SchedulerModuleImpl(
    private val ioScheduler: Scheduler,
    private val uiScheduler: Scheduler,
    private val computationScheduler: Scheduler
): SchedulerModule {

    override fun io() = ioScheduler

    override fun ui() = uiScheduler

    override fun computation() = computationScheduler
}

File: SharedCode/src/iosMain/swift/rxProxy/factories/SchedulerModuleImpl.swift
//
//  SchedulerModuleImpl.swift
//  ConceprFirstRxSwift
//
//  Created by Marc Gilbert on 06/12/2019.
//  Copyright © 2019 Marc Gilbert. All rights reserved.
//

import Foundation
import SharedCode
import RxSwift

class SchedulerModuleImpl: SharedCode.SchedulerModule {
    
    let ioScheduler = SchedulerImpl(schedulerType: ConcurrentDispatchQueueScheduler(qos: .background))
    let uiScheduler = SchedulerImpl(schedulerType: MainScheduler.instance)

    func io() -> SharedCode.Scheduler {
        return ioScheduler
    }
    
    func ui() -> SharedCode.Scheduler {
        return uiScheduler
    }
    
    func computation() -> SharedCode.Scheduler {
        return ioScheduler
    }
}
